var/spack/repos/builtin/packages/shamrock/package.py

```diff
@@ -0,0 +1,115 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Shamrock(CMakePackage):
+    """The Shamrock exascale framework for astrophysics"""
+
+    homepage = "https://shamrock-code.github.io/"
+    url = "https://github.com/Shamrock-code/Shamrock/releases/download/v2025.05.0/shamrock-2025.05.0.tar"
+    git = "https://github.com/Shamrock-code/Shamrock.git"
+
+    maintainers("tdavidcl")
+
+    license("CeCILL Free Software License Agreement v2.1", checked_by="tdavidcl")
+
+    version("main", branch="main", submodules=True)
+
+    version("2025.05.0", sha256="59d5652467fd9453a65ae7b48e0c9b7d4162edc8df92e09d08dcc5275407a897")
+
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    variant("shared", default=True, description="Enables the build of shared libraries")
+    variant("testing", default=True, description="Enables the build of shared libraries")
+    variant("pybindings", default=True, description="Install python bindings")
+
+    depends_on("sycl")
+    depends_on("mpi")
+    depends_on("python")
+    extends("python", when="+pybindings")
+
+    depends_on("ninja", type="build", when="generator=ninja")
+
+    conflicts("^intel-oneapi-runtime", msg="Shamrock supports it but this package does not yet.")
+
+    def cmake_args(self):
+        """Main configure step"""
+
+        spec = self.spec
+
+        args = ["-DSHAMROCK_ENABLE_BACKEND=SYCL"]
```

Comment:
I'd move the unconditional arguments first. Also I'd try to use the API provided by `CMakePackage` in the entire recipe. For instance, here I would do something similar to:
```suggestion
        args = [
            self.define("SHAMROCK_ENABLE_BACKEND", "SYCL"),
            self.define("PYTHON_EXECUTABLE", spec["python"].command.path),
            self.define_from_variant("BUILD_TEST", "testing"),
        ]
```